### ISSUE TITLE
Tooltip Wrong Calculation for Total Amount of items

### ISSUE THREAD

**Issue by Meisterzockerr** [Xruptor/BagSync#280]
In the new version you suddenly get not only you charactersname in the tooltip but on every char it shows now "Player".
Player shows all your items in your bags, bank and mail.
But for the Total amount it calculates the amount from you charname also.
Could we get a fix that removes the "Player" tag and only shows the Charname?
In this case it only shows Jessicalyn and not "Player".
![Bagsync tooltip](https://user-images.githubusercontent.com/128497870/226641635-ada35992-659c-4566-ab43-e294ee61497a.jpg)

Thanks in advance.

**Comment by Xruptor** [comment]
That's wrong and it's a bug.  It should not be showing a "player" at all.

Can you please upload here your BagSync.lua found under your WOW installation.  You may need to zip it up.  You can find it at the following location.
**WOW INSTALL Directory**\_retail_\WTF\Account\<your account name>\SavedVariables\**BagSync.lua**

**Comment by Meisterzockerr** [comment]
[BagSync.zip](https://github.com/Xruptor/BagSync/files/11033169/BagSync.zip)

**Comment by Meisterzockerr** [comment]
done.
I hope you can fix it :)
cant live without that addon. been doing so much AH stuff in wotlk.

**Comment by Xruptor** [comment]
> done. I hope you can fix it :) cant live without that addon. been doing so much AH stuff in wotlk.

Thank you! This will help tremendously to trace the issue 😃 .  I'm glad you enjoy the addon!  I try my best to keep it up to date and put new enhancements as I improve it.

**Comment by Xruptor** [comment]
I'm having a hard time replicating this issue.  This is especially true when the data you provided me doesn't match the tooltip you provided.  It may be outdated or updated versus what you gave me as BagSync.lua.

Example.  The picture you gave me tells me you have 76 Crystal Chalcedony Amulets in your bags.  However in the database you provided me, there are none in your bags and about 267 in the mailbox.

https://www.wowhead.com/wotlk/item=43245/crystal-chalcedony-amulet
ItemID: 43245

So I'm a bit perplexed XD.  I'm confused as to how "player" is showing up at all.  I even reverted back to 19.1 then 19.2 to see if I can replicate the problem.  Obviously it's a bug somewhere and I need to trace it, but I'm having a hard time tracing the issue.

Lets try a few things.

1) Go to BagSync Profiles /bgs profiles  and see if "player" is listed there.  If it's not then it's not stored in the Database and it's a tooltip issue.

2) Take a picture again of the issue happening.

3) Immediately without doing ANY changes to your bags or items.  Send me the BagSync.lua.  That way it's completely up to date and will match your picture.

You are playing on WOTLK correct?

**Comment by Xruptor** [comment]
In addition lets see if we can get this to trigger on your end as well. Make sure not to click on ANY items in your bags until instructed too.

type  **/reload** or **/reloadui** whatever you use to reload your UI
type  **/bgs debug**

- Enable the Debug
- Make sure only DEBUG, INFO and TRACE is selected.
- Press Clear on the bottom right
- Hover over an item that has the bug.  It should populate the Debug screen.
- Disable Debug
- Export the

**Comment by Meisterzockerr** [comment]
![Bagsync tooltip_2](https://user-images.githubusercontent.com/128497870/226866978-05d73db6-a454-48a7-a4f7-838b3137982f.jpg)

**Comment by Meisterzockerr** [comment]
[BagSync.zip](https://github.com/Xruptor/BagSync/files/11038693/BagSync.zip)

**Comment by Meisterzockerr** [comment]
i gonna try it with a different char since i DEd all the items :)
But i have the problem with any item on all of my chars

**Comment by Meisterzockerr** [comment]
[BagSync_with debug done.zip](https://github.com/Xruptor/BagSync/files/11038723/BagSync_with.debug.done.zip)

**Comment by Meisterzockerr** [comment]
![Bagsync tooltip_3](https://user-images.githubusercontent.com/128497870/226868613-803485c0-6405-4cdf-b88f-4645f1b3e147.jpg)

**Comment by Meisterzockerr** [comment]
I am playing Wotlk.
I use TSM and enabled/disabled the tooltip function but it had no effects.
The bug from bagsync didnt exist till the 19.x version.
all versions 18.xx and earlier didnt have that bug.

**Comment by Xruptor** [comment]
Yay! Thanks 😃.  Lets see if we can squash this bug together! The information you provided at least tells me it's not a database issue, so it has to be a display issue.  The question is... where the heck is it getting this "player" lol.  I'm determine to beat this bug with a stick until it's gone.

**Comment by Xruptor** [comment]
[BagSync.zip](https://github.com/Xruptor/BagSync/files/11040786/BagSync.zip)

Okay try that version of BagSync.  If you don't know how to install addons manually, just extract that to following directory.
`WOW INSTALL Directory\_retail_\Interace\Addons\`

Let me know if the "player" issue is still happening.  I think I know what it is, but I'm baffled as to why it's being returned by your server.

Please load /bgs debug if it's still doing the bug.

1. Only select, DEBUG, INFO, TRACE and SL2
2. clear the debug screen
3. hover over an item
4. export it here for me.

**Comment by Meisterzockerr** [comment]
seems fixed to me.
![Bagsync tooltip](https://user-images.githubusercontent.com/128497870/227018533-1069157a-0595-43e5-b092-2ce41b793451.jpg)

**Comment by Meisterzockerr** [comment]
the tooltip from 3 chars is from the TSM addon (will disable it, just for testing).
The bottom one with 2 chars is from Bagsync.
Doesnt show "player" anymore.
I'm curious what was the code that caused this display to happen? :)

**Comment by Meisterzockerr** [comment]
[BagSync_debugafterfix.zip](https://github.com/Xruptor/BagSync/files/11044171/BagSync_debugafterfix.zip)

**Comment by Xruptor** [comment]
Wait so it's fixed?  The numbers seem off to your TSM one as Indigowhite is missing?  Not sure what's up there because TGSM is showing it.

Also the bug was that UnitName() function from Blizzard was returning "player" when called prematurely.  So instead I told the addon to call it after everything is loaded to prevent "player" from happening.

Please let me know if the indigowhite thing is a bug I need to look into?

**Comment by Meisterzockerr** [comment]
All is fine. Indigo is one of my chars i logged on before.
I purged all BGS profiles and only logged into xjbank and Skylarvox.
So this is no bug or error.

**Comment by Xruptor** [comment]
Ah okay so it's fixed? Yay!
